server/controllers/payroll/employees/index.js

```diff
@@ -514,3 +515,97 @@ function find(options) {
 
   return db.exec(query, parameters);
 }
+
+/**
+ * @method patientToEmployee
+ *
+ * @description
+ * This function is responsible for transform a Patient to New employee in the database
+ */
+function patientToEmployee(req, res, next) {
+  const data = req.body;
+  const patientUuid = data.patient_uuid;
+  const employeeUuid = uuid();
+
+  data.creditor_uuid = uuid();
+  data.uuid = employeeUuid;
+
+  const employeeAdvantage = [];
+
+  // convert uuids to binary uuids as necessary
+  const employee = db.convert(data, [
+    'uuid', 'grade_uuid', 'debtor_group_uuid', 'creditor_group_uuid', 'creditor_uuid',
+    'debtor_uuid', 'current_location_id', 'origin_location_id', 'patient_uuid',
+  ]);
+
+  const employeeAdvantagePayroll = employee.payroll;
+
+  if (employeeAdvantagePayroll) {
+    Object.keys(employeeAdvantagePayroll).forEach((key) => {
+      employeeAdvantage.push([employee.uuid, key, employeeAdvantagePayroll[key]]);
+    });
+  }  
+
+  const creditor = {
+    uuid : employee.creditor_uuid,
+    group_uuid : employee.creditor_group_uuid,
+    text : `Crediteur [${employee.display_name}]`,
+  };
+
+  const debtor = {
+    uuid : employee.debtor_uuid,
+    group_uuid : employee.debtor_group_uuid,
+    text : `Debiteur [${employee.display_name}]`,
+  };
+
+  if (employee.date_embauche) {
+    employee.date_embauche = new Date(employee.date_embauche);
+  }
+
+  delete employee.debtor_group_uuid;
```

Comment:
Just FYI - there is a lodash function to remove lots of properties from an object called [_.omit()](https://lodash.com/docs/4.17.10#omit).

```js
_.omit(employee, ['debtor_group_uuid', 'creditor_group_uuid']);
```
```